pkgs/by-name/op/opera/package.nix

```
@@ -0,0 +1,145 @@
+{
+  alsa-lib,
+  atk,
+  cairo,
+  cups,
+  curl,
+  dbus,
+  dpkg,
+  expat,
+  fetchurl,
+  fontconfig,
+  freetype,
+  gdk-pixbuf,
+  glib,
+  gtk3,
+  gtk4,
+  lib,
+  libX11,
+  libxcb,
+  libXScrnSaver,
+  libXcomposite,
+  libXcursor,
+  libXdamage,
+  libXext,
+  libXfixes,
+  libXi,
+  libXrandr,
+  libXrender,
+  libXtst,
+  libdrm,
+  libnotify,
+  libpulseaudio,
+  libuuid,
+  libxshmfence,
+  libgbm,
+  nspr,
+  nss,
+  pango,
+  stdenv,
+  systemd,
+  at-spi2-atk,
+  at-spi2-core,
+  autoPatchelfHook,
+  wrapGAppsHook3,
+  qt6,
+  proprietaryCodecs ? false,
+  vivaldi-ffmpeg-codecs,
+}:
+
+let
+  mirror = "https://get.geo.opera.com/pub/opera/desktop";
+in
+stdenv.mkDerivation rec {
+  pname = "opera";
+  version = "121.0.5600.50";
+
+  src = fetchurl {
+    url = "${mirror}/${version}/linux/${pname}-stable_${version}_amd64.deb";
+    hash = "sha256-mUSUVWEdcSrwbWY8XFFAHrS7nagcXnNoLOqS8oL09Cc=";
+  };
+
+  nativeBuildInputs = [
+    dpkg
+    autoPatchelfHook
+    wrapGAppsHook3
+    qt6.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    alsa-lib
+    at-spi2-atk
+    at-spi2-core
+    atk
+    cairo
+    cups
+    curl
+    dbus
+    expat
+    fontconfig.lib
+    freetype
+    gdk-pixbuf
+    glib
+    gtk3
+    libX11
+    libXScrnSaver
+    libXcomposite
+    libXcursor
+    libXdamage
+    libXext
+    libXfixes
+    libXi
+    libXrandr
+    libXrender
+    libXtst
+    libdrm
+    libnotify
+    libuuid
+    libxcb
+    libxshmfence
+    libgbm
+    nspr
+    nss
+    pango
+    (lib.getLib stdenv.cc.cc)
+    qt6.qtbase
+  ];
+
+  runtimeDependencies = [
+    # Works fine without this except there is no sound.
+    libpulseaudio.out
+
+    # This is a little tricky. Without it the app starts then crashes. Then it
+    # brings up the crash report, which also crashes. `strace -f` hints at a
+    # missing libudev.so.0.
+    (lib.getLib systemd)
+
+    # Error at startup:
+    # "Illegal instruction (core dumped)"
+    gtk3
+    gtk4
+  ]
+  ++ lib.optionals proprietaryCodecs [
+    vivaldi-ffmpeg-codecs
+  ];
+
+  dontWrapQtApps = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -r usr $out
+    cp -r usr/share $out/share
+    # we already using QT6, autopatchelf wants to patch this as well
+    rm $out/usr/lib/x86_64-linux-gnu/opera/libqt5_shim.so
+    ln -s $out/usr/bin/opera $out/bin/opera
+  '';
+
+  meta = with lib; {
```

```suggestion
  meta = {
```
`with` before attrsets (especially `with lib`) was used and abused to death and is now frowned upon

---

Addressed.